helper/funcs.go

```diff
@@ -716,3 +717,21 @@ func NewSafeTimer(duration time.Duration) (*time.Timer, StopFunc) {
 
 	return t, cancel
 }
+
+// IsMethodHTTP returns whether s is a known HTTP method, ignoring case.
+func IsMethodHTTP(s string) bool {
+	switch strings.ToUpper(s) {
+	case http.MethodGet:
+	case http.MethodHead:
+	case http.MethodPost:
+	case http.MethodPut:
+	case http.MethodPatch:
+	case http.MethodDelete:
+	case http.MethodConnect:
+	case http.MethodOptions:
+	case http.MethodTrace:
+	default:
+		return false
+	}
+	return true
```

Comment:
This is just a note rather than a suggestion for other readers as this made me look into the difference between this style and having a single case to catch all the methods like below.

```
func IsMethodHTTP_ALT(s string) bool {
	switch strings.ToUpper(s) {
	case http.MethodGet, http.MethodHead, http.MethodPost, http.MethodPut, http.MethodPatch,
		http.MethodDelete, http.MethodConnect, http.MethodOptions, http.MethodTrace:
		return true
	default:
		return false
	}
}
```

I ran a benchmark as I was curious if there was a difference and wonder if this will change with the changes made in go 1.19 to the compiler using a jump table for string switch statements.
```
Benchmark_IsMethodHTTP-16        	134591882	         9.000 ns/op
Benchmark_IsMethodHTTP_ALT-16    	163068037	         7.306 ns/op
```

Anyway, please ignore, I was just curious and the current implementation is slightly easier to read and 2ns is nothing. :D

Reply:
Always love a good benchmark! FWIW I tend to code more vertically because that's the only direction my mousewheel goes :disappointed: